src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -252,6 +281,38 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
                 response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
             }
 
+        } else if (!strcmp(commands_list[WM_TASK_UPGRADE_RESULT], command)) {
```

Comment:
Is there any way we can reduce code complexity on `wm_task_manager_analyze_task`? otherwise it will be really hard to do unit tests on it.